client/pkg/omnictl/jointoken.go

```diff
@@ -0,0 +1,274 @@
+			}
+
+			return access.WithClient(func(ctx context.Context, client *client.Client) error {
+				_, err := safe.StateUpdateWithConflicts(
+					ctx,
+					client.Omni().State(),
+					siderolink.NewJoinToken(resources.DefaultNamespace, id).Metadata(),
+					func(res *siderolink.JoinToken) error {
+						res.TypedSpec().Value.ExpirationTime = timestamppb.New(time.Now().Add(joinTokenRenewFlags.ttl))
+
+						return nil
+					},
+				)
+
+				if err != nil {
+					return err
+				}
+
+				fmt.Printf("token %q was renewed, new ttl is %s\n", id, joinTokenRenewFlags.ttl)
+
+				return nil
+			})
+		},
+	}
+
+	joinTokenListCmd = &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"l"},
+		Short:   "List join tokens",
+		Args:    cobra.NoArgs,
+		RunE: func(*cobra.Command, []string) error {
+			return access.WithClient(func(ctx context.Context, client *client.Client) error {
+				joinTokens, err := safe.ReaderListAll[*siderolink.JoinTokenStatus](ctx, client.Omni().State())
+				if err != nil {
+					return err
+				}
+
+				writer := tabwriter.NewWriter(os.Stdout, 0, 0, 3, ' ', 0)
+
+				fmt.Fprintf(writer, "ID\tNAME\tSTATE\tEXPIRATION\tUSE COUNT\tDEFAULT\n") //nolint:errcheck
+
+				for token := range joinTokens.All() {
+					var isDefault string
+
+					if token.TypedSpec().Value.IsDefault {
+						isDefault = "*"
+					}
+
+					expirationTime := "never"
+
+					if token.TypedSpec().Value.ExpirationTime != nil {
+						expirationTime = token.TypedSpec().Value.ExpirationTime.AsTime().String()
+					}
+
+					fmt.Fprintf(
+						writer,
+						"%s\t%s\t%s\t%s\t%d\t%s\n",
+						token.Metadata().ID(),
+						token.TypedSpec().Value.Name,
+						token.TypedSpec().Value.State.String(),
+						expirationTime,
+						token.TypedSpec().Value.UseCount,
+						isDefault,
+					)
+				}
+
+				return writer.Flush()
+			})
+		},
+	}
+
+	joinTokenDeleteCmd = &cobra.Command{
+		Use:     "delete <name>",
+		Aliases: []string{"d"},
+		Short:   "Delete a join token",
+		Args:    cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			id := args[0]
+
+			return access.WithClient(func(ctx context.Context, client *client.Client) error {
+				err := client.Omni().State().TeardownAndDestroy(ctx, siderolink.NewJoinToken(resources.DefaultNamespace, id).Metadata())
+				if err != nil {
+					return fmt.Errorf("failed to delete a join token: %w", err)
+				}
+
+				fmt.Printf("deleted join token: %s\n", id)
+
+				return nil
+			})
+		},
+	}
+)
+
+func init() {
+	RootCmd.AddCommand(joinTokenCmd)
+
+	joinTokenCmd.AddCommand(joinTokenCreateCmd)
+	joinTokenCmd.AddCommand(joinTokenListCmd)
+	joinTokenCmd.AddCommand(joinTokenDeleteCmd)
+	joinTokenCmd.AddCommand(joinTokenRevokeCmd)
+	joinTokenCmd.AddCommand(joinTokenMakeDefaultCmd)
+	joinTokenCmd.AddCommand(joinTokenUnrevokeCmd)
+	joinTokenCmd.AddCommand(joinTokenRenewCmd)
+
+	joinTokenCreateCmd.Flags().DurationVarP(&joinTokenCreateFlags.ttl, "ttl", "t", 0, "TTL for the join token")
```

Comment:
is zero reasonable default here?

Reply:
yeah. 0 will make this token never expire

Reply:
I think it's fine there too as with 0 it will fail the command and then the user will have to set some value